pkg/api/graph/graphview/veneering_test.go

```diff
@@ -259,38 +259,35 @@ func TestGraph(t *testing.T) {
 	})
 	deploygraph.EnsureDeploymentConfigNode(g, &deployapi.DeploymentConfig{
 		ObjectMeta: kapi.ObjectMeta{Namespace: "other", Name: "deploy1"},
-		Triggers: []deployapi.DeploymentTriggerPolicy{
-			{
-				ImageChangeParams: &deployapi.DeploymentTriggerImageChangeParams{
-					From:           kapi.ObjectReference{Namespace: "default", Name: "other"},
-					ContainerNames: []string{"1", "2"},
-					Tag:            "tag1",
+		Spec: deployapi.DeploymentConfigSpec{
+			Triggers: []deployapi.DeploymentTriggerPolicy{
+				{
+					ImageChangeParams: &deployapi.DeploymentTriggerImageChangeParams{
+						From:           kapi.ObjectReference{Kind: "ImageStreamTag", Namespace: "default", Name: "other:tag1"},
```

Comment:
> I guess alternatively, was this simply an invalid construct that could never have been created in the external version of the API?

Yes, we don't expose Tag and RepositoryName. Furthermore, answering your 1st question, if the object reference here is an ImageStream (kind is missing from this test but generally [defaulting](https://github.com/openshift/origin/blob/0d4de0b1801e8c58a28c8bcd9d12b5d0c46de1cb/pkg/deploy/api/v1/conversion.go#L294) would set it to an ImageStreamTag, [validation](https://github.com/openshift/origin/blob/0d4de0b1801e8c58a28c8bcd9d12b5d0c46de1cb/pkg/deploy/api/validation/validation.go#L259) would set it to an ImageStream, still not confused? :) ) then yes, it would be converted like [this](https://github.com/openshift/origin/blob/0d4de0b1801e8c58a28c8bcd9d12b5d0c46de1cb/pkg/deploy/api/v1/conversion.go#L176-L182).